lib/dpt/types.js

```diff
@@ -0,0 +1,198 @@
+const ip = require('ip')
+const rlp = require('rlp')
+const secp256k1 = require('secp256k1')
+const keccak256 = require('../util').keccak256
+
+const TYPENAMES = {
+  BY_NAME: {
+    ping: 0x01,
+    pong: 0x02,
+    findneighbours: 0x03,
+    neighbours: 0x04
+  },
+  BY_TYPE: {
+    0x01: 'ping',
+    0x02: 'pong',
+    0x03: 'findneighbours',
+    0x04: 'neighbours'
+  }
+}
+
+function getTimestamp () {
+  return (Date.now() / 1000) | 0
+}
+
+const timestamp = {
+  encode: function (value) {
+    if (value === undefined) value = getTimestamp() + 60 // hack
+    let buffer = new Buffer(4)
+    buffer.writeUInt32BE(value)
+    return buffer
+  },
+  decode: function (buffer) {
+    if (buffer.length !== 4) throw new RangeError(`Invalid timestamp buffer :${buffer.toString('hex')}`)
+    return buffer.readUInt32BE(0)
+  }
+}
+
+const address = {
+  encode: function (value) {
+    if (ip.isV4Format(value)) return ip.toBuffer(value)
+    if (ip.isV6Format(value)) return ip.toBuffer(value)
+    throw new Error(`Invalid address: ${value}`)
+  },
+  decode: function (buffer) {
+    if (buffer.length === 4) return ip.toString(buffer)
+    if (buffer.length === 16) return ip.toString(buffer)
+    throw new Error(`Invalid address buffer: ${buffer.toString('hex')}`)
```

Comment:
so after looking more closely at the code; it shouldn't be RLP encoded at this point. I'm not sure what can explain the lenght of 12